docs/BaremetalHost_ProvisioningState.dot

```diff
@@ -40,11 +40,17 @@ digraph BaremetalHost {
     Deleting5 [shape=point]
 
     Ready [shape=doublecircle]
-    Ready -> Provisioning [label="NeedsProvisioning()"]
+    Ready -> Preparing [label="NeedsProvisioning()"]
     Ready -> Deleting6 [label="!DeletionTimestamp.IsZero()"]
 
     Deleting6 [shape=point]
 
+    Preparing -> Prepared [label=done]
+    Preparing -> Deleting [label="preparing error"]
```

Comment:
Deleting the host because you can't provision is definitely, definitely not what you want to do. Probably it should just return to the Ready state.

It's correct that there is a transition to Deleting though, but only when `!DeletionTimestamp.IsZero()`. (Also the diagram might end up tidier if you use the same Deleting6 state as the target the Ready uses.)

Reply:
But if this error isn't temporary, it will cause an endless loop of `Ready`-> `Preparing`-> `Ready`.

Reply:
Currently we respond to permanent failures by setting an Error and not rescheduling another reconcile (so the resource won't be reconciled again until it changes). So you can do that.
However recent discussion has tended toward the theory that we ought to just retry in an endless loop anyway.